src/games/strategy/triplea/TripleA.java

```diff
@@ -115,18 +117,25 @@ public void startGame(final IGame game, final Set<IGamePlayer> players, final bo
       connectPlayers(players, null);
     } else {
       SwingAction.invokeAndWait(() -> {
-        final TripleAFrame frame;
-        frame = new TripleAFrame(game, localPlayers);
-        display = new TripleADisplay(frame);
-        game.addDisplay(display);
-        soundChannel = new DefaultSoundChannel(localPlayers);
-        game.addSoundChannel(soundChannel);
-        frame.setSize(700, 400);
-        frame.setVisible(true);
-        ClipPlayer.play(SoundPath.CLIP_GAME_START);
-        connectPlayers(players, frame);
-        frame.setExtendedState(Frame.MAXIMIZED_BOTH);
-        frame.toFront();
+        try {
+          final TripleAFrame frame;
+          frame = new TripleAFrame(game, localPlayers);
+          display = new TripleADisplay(frame);
+          game.addDisplay(display);
+          soundChannel = new DefaultSoundChannel(localPlayers);
+          game.addSoundChannel(soundChannel);
+          frame.setSize(700, 400);
+          frame.setVisible(true);
+          ClipPlayer.play(SoundPath.CLIP_GAME_START);
+          connectPlayers(players, frame);
+          frame.setExtendedState(Frame.MAXIMIZED_BOTH);
+          frame.toFront();
+        } catch (Exception e) {
+          if(game instanceof ClientGame){
```

Comment:
`instanceof` checks are always a major design smell. There are generally better ways to accomplish the same, notably via polymorphism. But more to point, if a caller passed in an exception action, it probably wants it to be executed. In addition, we call ClientLogger in all types of code, it defaults to printing to System.out if there is no Swing UI to print to. It wouldn't hurt to verify this, but ClientLogger should be safe to call in just about  any context.